src/components/pages/TransactionHistoryPage.js

```diff
@@ -1,34 +1,106 @@
-import { React, useState, useEffect } from "react";
+import { React, useState, useEffect, useContext, useCallback } from "react";
 import Navbar from "../reusable/NavBar";
 import Sidebar from "../reusable/SideBar";
+import AlgoquantApiContext from "../../api/ApiContext";
+import Table from "../reusable/Table";
+import formatter from "../utils/CurrencyFormatter";
+import { TableSpinner } from "../reusable/LoadSpinner";
 
+const FETCH_AMOUNT = 10;
 const TransactionHistoryPage = () => {
+  const [history, setHistory] = useState([]);
+  const [lastPage, setLastPage] = useState(false);
   const [transactions, setTransactions] = useState([]);
   const [page, setPage] = useState(1);
+  // State variables used to access algoquant SDK API and display/ keep state of user data from database
+  const algoquantApi = useContext(AlgoquantApiContext);
+  const [lastKey, setLastKey] = useState(null);
+  const [pagesSeen, setPagesSeen] = useState(0);
+  const [isLoading, setIsLoading] = useState(true);
+
+  const fetchTrades = useCallback(
+    (FETCH_AMOUNT) => {
+      const historyBuffer = [];
+      if (!lastPage && pagesSeen < page) {
+        if (algoquantApi.token) {
+          algoquantApi
+            .getTrades(FETCH_AMOUNT, lastKey)
+            .then((resp) => {
+              setLastKey(null);
+              setPagesSeen(pagesSeen + 1);
+              if (resp.data.LastEvaluatedKey === undefined && page !== 1) {
```

Comment:
Why do you need to check if the page !== 1 ?

Reply:
the last query should never have a "last evaluated key"  and the page !== 1 makes sure i set set the state to true once we have fetched the last query aka that would mean we are querying for the last page. and that state variable is used to keep track of when we should stop querying for new data to stop populating our local storage of trades

Reply:
nvm i removed